src/chains/algorand_1/algoCreateAccount.ts

```diff
@@ -31,19 +39,22 @@ export class AlgorandCreateAccount implements CreateAccount {
   constructor(chainState: AlgorandChainState, options?: AlgorandCreateAccountOptions) {
     this._chainState = chainState
     this._options = options
-    const multiSigOptions = this?._options?.multiSigOptions
-    // if multisig options are given, then compute a multisig account address using the passed in algorand addresses in multisig options
-    if (multiSigOptions) {
-      this._accountName = determineMultiSigAddress(multiSigOptions)
-    }
+    this._publicKey = options?.publicKey
+    this._multiSigOptions = options?.multiSigOptions || null
   }
 
   // ---- Interface implementation
 
   /** Account name for the account to be created
    *  May be automatically generated (or otherwise changed) by composeTransaction() */
-  get accountName(): any {
-    return this._accountName
+  get accountName(): AlgorandEntityName {
+    if (this._publicKey) {
+      return toAlgorandEntityName(toAddressFromPublicKey(this._publicKey))
+    }
+    if (this._multiSigOptions) {
```

Comment:
this _multiSigOptions if block should be before the public key if block. If we have multiSigOptions, the accountName is always based off that. As written, this could break multisig